### PR TITLE
Add local_dir

### LINKED ISSUE #55: Pass local_dir parameter to search functions
**Problem:** It doesn't appear possible to pass the `local_dir` parameter to the search functions in `tune-sklearn`. This causes the temporary files to be created in the default directory, which in my case is problematic due to file size/count constraints on my home directory. 

**Proposed solution:** Allow the `local_dir` parameter to be passed to the functions, for example:

```python
tune_search = TuneGridSearchCV(
    pipeline_svc,
    parameters,
    local_dir = '/.../',   # <-----
    max_iters=10, 
)
``` 

There might be an alternative method to achieve this that is already implemented, however, I was going by the following part of the Ray documentation: https://docs.ray.io/en/latest/tune/api_docs/logging.html?highlight=ray_results#log-directory

Thanks!

### PR DESCRIPTION
**Pull request by inventormc**
Resolves #55

### CODE REVIEW

File: tests/test_randomizedsearch.py [hunk 1]
```diff
@@ -84,6 +88,28 @@ def test_check_cv_results_keys(cv_results, param_keys, score_keys,
...
+        tune_search = TuneSearchCV(
+            clf,
+            parameter_grid,
+            early_stopping=scheduler,
+            max_iters=10,
+            local_dir="./test-result"
+        )
+        tune_search.fit(x, y)
+
+        self.assertTrue(len(os.listdir("./test-result")) != 0)
```
Comment (richardliaw) [inline tests/test_randomizedsearch.py:110]: nice!!

File: tune_sklearn/tune_search.py [hunk 2]
```diff
@@ -305,7 +312,8 @@ def _tune_run(self, config, resources_per_trial):
                     config=config,
                     fail_fast=True,
                     checkpoint_at_end=True,
-                    resources_per_trial=resources_per_trial)
+                    resources_per_trial=resources_per_trial,
+                    local_dir=self.local_dir)
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:316]:
```suggestion
                    local_dir=os.path.expanduser(self.local_dir))
```

File: tune_sklearn/tune_search.py [hunk 3]
```diff
@@ -317,7 +325,8 @@ def _tune_run(self, config, resources_per_trial):
...
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:329]: expanduser here

File: tune_sklearn/tune_search.py [hunk 4]
```diff
@@ -336,6 +345,7 @@ def _tune_run(self, config, resources_per_trial):
                 config=config,
                 fail_fast=True,
                 checkpoint_at_end=True,
-                resources_per_trial=resources_per_trial)
+                resources_per_trial=resources_per_trial,
+                local_dir=self.local_dir)
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:349]: exapnduser here

File: tune_sklearn/tune_gridsearch.py [hunk 5]
```diff
@@ -205,6 +213,7 @@ def _tune_run(self, config, resources_per_trial):
                 config=config,
                 fail_fast=True,
                 checkpoint_at_end=True,
-                resources_per_trial=resources_per_trial)
+                resources_per_trial=resources_per_trial,
+                local_dir=self.local_dir)
```
Comment (richardliaw) [inline tune_sklearn/tune_gridsearch.py:217]: expanduser

File: tune_sklearn/tune_gridsearch.py [hunk 6]
```diff
@@ -194,7 +201,8 @@ def _tune_run(self, config, resources_per_trial):
...
```
Comment (richardliaw) [inline tune_sklearn/tune_gridsearch.py:205]: expanduser

**Comment by anthonyhsyu** [comment]
Memory leaks (probably fixed by #61?) + lint then we can merge this

**Pull request merged**